### PR TITLE
[CUR-330] Adding subject tags filters based on discipline tags in CAT

### PR DESCRIPTION
**Pull request by BTitterington**
## Description

CUR-330 - https://www.notion.so/oaknationalacademy/Add-subject-tags-for-Science-years-7-8-9-e65d572090ad43e78c5e0a76bf2fae06

This feature adds science subject tabs for KS3 (Years 7, 8 & 9) to allow users to filter and find relevant units.

They have been added in CAT as tags on the units with the category type "Discipline"

An MV was created to bring in the tag information for each unit, and this ticket uses those tags to add the filters/tabs when a unit has those attached.

NOTE: This is a new branch and PR for this as I had a git issue.

## How to test

1. Go to https://deploy-preview-2340--oak-web-application.netlify.thenational.academy/teachers/curriculum/science-secondary-aqa/units
2. You should see these tabs on year 7 - All, Biology, Chemistry & Physics
3. Selecting these should filter the units as expected (check expected results in CAT)

## Screenshots

**How it used to look:**

image
 
**How it should now look:**

image

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx [hunk 1]
```diff
@@ -65,26 +74,50 @@ type CurriculumVisualiserProps = {
...
+    unit.tier === null &&
+    unit.subject_parent === null &&
+    duplicateUnitSlugs.has(unit.slug);
+  return (
+    filterBySubject &&
+    filterByDomain &&
+    filterByTier &&
+    filterByDiscipline &&
+    !isDuplicate
+  );
```
Comment (BTitterington) [inline src/components/CurriculumComponents/CurriculumVisualiser/CurriculumVisualiser.tsx:120]: It's not made this easy to see, but this is effectively the removal of createProgrammeSlug (It's already in UnitsTab) and the export of isVisibleUnit (so the duplication of that function can be removed from UnitsTab)

File: src/components/CurriculumComponents/UnitsTab/UnitsTab.tsx [hunk 2]
```diff
@@ -214,11 +231,21 @@ const UnitsTab: FC<UnitsTabProps> = ({ data, examboardSlug }) => {
         });
       }
       filters.tiers.sort((a, b) => a.tier_slug.localeCompare(b.tier_slug));
+      // Sort disciplines
+      filters.disciplines.sort((a, b) => a.title.localeCompare(b.title));
+
+      // Add an "All" option if there are 2 or more disciplines. Set to -1 id as this shouldn't ever appear in the DB
+      const allDisciplineTag: Discipline = { id: -1, title: "All" };
```
Comment (sonalisin) [inline src/components/CurriculumComponents/UnitsTab/UnitsTab.tsx:238]: Why is it greater than 2 and not 1? Will there be a default of 2 if there are 2, or are there no single disciplines

**Review by sonalisin — APPROVED** [review]
Looks good! 😄

File: src/components/CurriculumComponents/UnitsTab/UnitsTab.tsx (hunk 2 above)
Comment (BTitterington) [reply src/components/CurriculumComponents/UnitsTab/UnitsTab.tsx:238]: Greater than or equal to 2. So, my theory was that there is no point showing the filter unless there are at least 2 different disciplines to choose from.

Comment (sonalisin) [reply src/components/CurriculumComponents/UnitsTab/UnitsTab.tsx:238]: Ah, got you. Makes sense 👍

**Pull request merged**